tests/e2e/performance/test_pvc_creation_deletion_performance.py

```diff
@@ -0,0 +1,291 @@
+                    f"{accepted_deletion_time} seconds."
+                )
+
+            deletion_time_measures.append(deletion_time)
+
+        self.process_measurements(
+            creation_time_measures, deletion_time_measures, msg_prefix
+        )
+
+    def process_measurements(
+        self, creation_time_measures, deletion_time_measures, msg_prefix
+    ):
+        """
+           Analyses the PVC creation and deletion times. If these times, for both creation and deletion, are within
+           the given limits and the standard deviation is smaller than the predefined accepted one,
+           writes them to the codespeed dashboard. Otherwise, fails the test
+        Args:
+            creation_time_measures: A list of PVC creation time measurements
+            deletion_time_measures: A list of PVC deletion time measurements
+            msg_prefix: A string for comprehensive logging
+
+        """
+        accepted_creation_deviation_percent = 50
+        accepted_deletion_deviation_percent = 50
+
+        creation_average = statistics.mean(creation_time_measures)
+        log.info(
+            f"{msg_prefix} The average creation time for the sampled {len(creation_time_measures)} "
+            f"PVCs is {creation_average} seconds."
+        )
+
+        deletion_average = statistics.mean(deletion_time_measures)
+        log.info(
+            f"{msg_prefix}The average deletion time for the sampled {len(deletion_time_measures)} PVCs "
+            f"is {deletion_average} seconds."
+        )
+
+        creation_st_deviation = statistics.stdev(creation_time_measures)
+        creation_st_deviation_percent = creation_st_deviation / creation_average * 100.0
+        if creation_st_deviation_percent > accepted_creation_deviation_percent:
+            raise ex.PerformanceException(
+                f"{msg_prefix} PVC creation time deviation is {creation_st_deviation_percent}% "
+                f"and is greater than the allowed {accepted_creation_deviation_percent}%."
+            )
+
+        log.info(
+            f"{msg_prefix} The standard deviation percent for creation of {len(deletion_time_measures)} sampled "
+            f"PVCs is {creation_st_deviation_percent}%."
+        )
+
+        deletion_st_deviation = statistics.stdev(deletion_time_measures)
+        deletion_st_deviation_percent = deletion_st_deviation / deletion_average * 100.0
+        if deletion_st_deviation_percent > accepted_deletion_deviation_percent:
+            raise ex.PerformanceException(
+                f"{msg_prefix} PVC deletion time deviation is {deletion_st_deviation_percent}%"
+                f"and is greater than the allowed {accepted_deletion_deviation_percent}%."
+            )
+
+        log.info(
+            f"{msg_prefix} The standard deviation percent for deletion of {len(deletion_time_measures)} sampled PVCs "
+            f"is {deletion_st_deviation_percent}%."
+        )
```

Comment:
instead of duplicate the code for creation and deletion, you can create a function which accept (values, title), the title is create/delete.

Reply:
@Avilir  I'll try to implement such a function.

Reply:
@Avilir  Done. 
def process_time_measurements( self, action_name, time_measures, accepted_deviation_percent, msg_prefix ) was implemented.